src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeEntryProcessor.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewFieldBuilder;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.types.BaseType;
+import uk.gov.hmcts.ccd.domain.types.CollectionValidator;
+import uk.gov.hmcts.ccd.endpoint.exceptions.DataProcessingException;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.*;
+import static uk.gov.hmcts.ccd.domain.service.processor.DisplayContextParameter.*;
+
+@Component
+public class DateTimeEntryProcessor extends CaseDataFieldProcessor {
```

Comment:
Important:
we are introducing the concept of a generic FieldProcessor which is good and in line with the ticket implementation suggestion. I've been wanting to implement this for long. 
But because this is meant to be generic so that can be used in the future for other types of processing, we cannot use inheritance but instead we should use composition. Otherwise we'll end up with an exponentially growing number of subclasses when we need to introduce other types of processing. And it will be not possible to combine different types of processing together. For example CRUD processing together with DateTime formatting. 
We can keep this for now. But I will raise a separate redesign ticket to introduced a generic CaseFieldProcessor.